mongo_connector/doc_managers/elastic_doc_manager.py

```diff
@@ -118,11 +137,20 @@ def update(self, document_id, update_spec, namespace, timestamp):
         """
         self.commit()
         index, doc_type = self._index_and_mapping(namespace)
-        document = self.elastic.get(index=index, doc_type=doc_type,
-                                    id=u(document_id))
-        updated = self.apply_update(document['_source'], update_spec)
-        # _id is immutable in MongoDB, so won't have changed in update
-        updated['_id'] = document['_id']
+
+        document = None
+        if doc_type in self.routing:
+            # we need to manually retrieve the parent id from mongo
+            db, collection = self.command_helper.revert_namespace(namespace)
+            updated = self.mongo[db][collection].find_one(document_id)
+            if updated is None:
+                return # if the parent was deleted we can't do anything about it
```

Comment:
Just added a comment for this to the source too for later reference but:

You can't do any operations (`get`, `update`, etc.) in elasticsearch without the full routing to a document, and that includes the parent key. Since the delta document we get from the oplog might not contain the parent id, we have to retrieve the document ourselves.

The `doc_type in self.routing` (added `'parentPath' in self.routing[doc_type]` just now too) signifies that this is a parent-child relationship.

We could do a `_search` for the document instead of the Mongo request, not sure which would be more preferable.